include/deal.II/base/parameter_acceptor.h

```diff
@@ -0,0 +1,528 @@
+ *   ...
+ *   end
+ * end
+ * subsection Class B
+ *   subsection Class
+ *   ...
+ *   end
+ *   subsection Forcing term
+ *   ...
+ *   end
+ * end
+ * @endcode
+ *
+ * Note the "/" at the begin of the string name. This is interpreted by
+ * ParameterAcceptor like the root folder in Unix systems. This means that the
+ * sections "Class A" and "Class B" will not be nested under any section. On
+ * the other hand, if the string does not begin with a "/" as in the previous
+ * cases the section will be created **under the current path**, which depends
+ * on the previously defined sections/subsections/subsubsections. Indeed, the
+ * section "Forcing term" is nested under "Class A" or "Class B". To make
+ * things more clear. let's consider the following two examples
+ *
+ * @code
+ * int main()
+ * {
+ *  MyClass ca("/Class A/Class");
+ *  MyClass cb("Class B/Class");
+ *  ParameterAcceptor::initialize("file.prm");
+ * }
+ * @endcode
+ * The parameter file will have the following structure
+ * @code
+ * subsection Class A
+ *   subsection Class
+ *   ...
+ *   end
+ *   subsection Forcing term
+ *   ...
+ *   end
+ *   subsection Class B
+ *     subsection Class
+ *     ...
+ *     end
+ *     subsection Forcing term
+ *     ...
+ *     end
+ *   end
+ * end
+ * @endcode
+ *
+ * If instead one of the paths ends with "/" instead of just
+ * a name of the class, subsequent classes will interpret this as
+ * a full path, interpreting the class name as a directory name:
+ * @code
+ * int main()
+ * {
+ *  MyClass ca("/Class A/Class/");
+ *  MyClass cb("Class B/Class");
+ *  ParameterAcceptor::initialize("file.prm");
+ * }
+ * @endcode
+ * The parameter file will have the following structure
+ * @code
+ * subsection Class A
+ *   subsection Class
+ *      ...
+ *      subsection Forcing term
+ *      ...
+ *      end
+ *      subsection Class B
+ *          subsection Class
+ *          ...
+ *          end
+ *          subsection Forcing term
+ *          ...
+ *          end
+ *      end
+ *   end
+ * end
+ * @endcode
+ *
+ * As a final remark, in order to allow a proper management of all the
+ * sections/subsections, the instantiation of objects and the call to
+ * ParameterAcceptor::initialize() **cannot be done in multithread**.
+ *
+ * If you pass an empty name, the boost::core::demangle() function is used to
+ * fill the section name with a human readable version of the class name
+ * itself.
+ *
+ * @author Luca Heltai, 2017.
+ */
+class ParameterAcceptor : public Subscriptor
+{
+public:
+  /**
+   * The constructor adds derived classes to the list of acceptors. If
+   * a section name is specified, then this is used to scope the
+   * parameters in the given section, otherwise a pretty printed
+   * version of the derived class is used.
+   */
+  ParameterAcceptor(const std::string section_name="");
+
+  /**
+   * The destructor sets to zero the pointer relative to this index,
+   * so that it is safe to destroy the mother class.
```

Comment:
I think that's an implementation detail, so I would omit it.

It's also not the correct order -- the derived class is destroyed *first*, only then comes the base class. You can't change this, and it doesn't matter as long as you're not trying to use pointers in the base class that point to the derived class. I just wouldn't mention it at all.